src/js/utils/specs/ajax-helpers-spec.js

```diff
@@ -0,0 +1,166 @@
+                        $.ajax(testUrl);
+                        AjaxHelpers.expectRequest(requests, 'GET', testUrl, undefined);
+                    }));
+
+                    it('verifies a more complex request', AjaxHelpers.withFakeRequests(function(requests) {
+                        $.ajax(testUrl, {
+                            method: 'POST',
+                            data: 'foobar'
+                        });
+                        AjaxHelpers.expectRequest(requests, 'POST', testUrl, 'foobar');
+                    }));
+                });
+
+                describe('expectNoRequests', function() {
+                    it('verifies 0 requests when none are made', AjaxHelpers.withFakeRequests(function(requests) {
+                        AjaxHelpers.expectNoRequests(requests);
+                    }));
+
+                    it('verifies 0 requests when all are handled', AjaxHelpers.withFakeRequests(function(requests) {
+                        $.ajax(testUrl);
+                        AjaxHelpers.respondWithError(requests);
+                        AjaxHelpers.expectNoRequests(requests);
+                    }));
+                });
+
+                describe('expectJsonRequest', function() {
+                    it('validates a request with JSON payload', AjaxHelpers.withFakeRequests(function(requests) {
+                        $.ajax(testUrl, {
+                            method: 'POST',
+                            data: JSON.stringify(testData),
+                            contentType: 'application/json'
+                        });
+                        AjaxHelpers.expectJsonRequest(requests, 'POST', testUrl, testData);
+                    }));
+                });
+
+                describe('expectRequestURL', function() {
+                    it('validates a JSON request made to a URL', AjaxHelpers.withFakeRequests(function(requests) {
+                        var path = '/home/category';
+
+                        $.ajax(testUrl + path + '?' + testQuerystring);
+                        AjaxHelpers.expectRequestURL(requests, path, testData);
+                    }));
+                });
+
+                describe('expectPostRequest', function() {
+                    it('validates a non-JSON POST request', AjaxHelpers.withFakeRequests(function(requests) {
+                        $.ajax(testUrl, {
+                            method: 'POST',
+                            data: testQuerystring
+                        });
+                        AjaxHelpers.expectPostRequest(requests, testUrl, testQuerystring);
+                    }));
+                });
+
+                describe('skipResetRequest', function() {
+                    it('verifies request was cancelled', AjaxHelpers.withFakeRequests(function(requests) {
+                        $.ajax(testUrl).abort();
+                        AjaxHelpers.skipResetRequest(requests);
```

Comment:
I agree, I'm not a bug fan of the expectation helper pattern for this reason (because it makes it hard to see what's happening in tests). I'll add a line that checks to see if currentIndex is incremented, though, that's a good idea.